src/Context/Context.php

```diff
@@ -4,111 +4,35 @@
-     *
-     * In practical terms, this is implemented as a linked list of Context instances, with each one holding a reference
-     * to the key object, the value that corresponds to the key, and an optional reference to the parent Context
-     * (i.e. the next link in the linked list chain)
-     *
-     * @param ContextKey|null $key The key object. Should only be null when creating an "empty" context
-     * @param mixed $value
-     * @param Context|null $parent Reference to the parent object
-     */
-    private function __construct(?ContextKey $key=null, $value=null, ?self $parent=null)
-    {
-        $this->key = $key;
-        $this->value = $value;
-        $this->parent = $parent;
-    }
-
-    /**
-     * Static version of get()
-     * This is primarily useful when the caller doesn't already have a reference to the Context that they want to mutate.
-     * This will operate on the "current" global context in that scenario.
-     *
-     * There are two ways to call this function:
-     * 1) With a $context value:
-     *    Context::getValue($key, $context) is functionally equivalent to $context->get($key)
-     * 2) Without a $context value:
-     *    This will fetch the "current" Context if one exists or create one if not, then attempt to get the value from it.
-     *
-     * @param ContextKey $key
-     * @param Context|null $context
-     *
-     * @return mixed
-     */
-    public static function getValue(ContextKey $key, ?self $context=null)
-    {
-        $context ??= self::getCurrent();
 
-        return $context->get($key);
-    }
-
-    /**
-     * This adds a key/value pair to this Context. We do this by instantiating a new Context instance with the key/value and pass
-     * a reference to $this as the "parent" creating the linked list chain.
-     *
-     * @param ContextKey $key
-     * @param mixed $value
-     *
-     * @return Context a new Context containing the key/value
-     */
-    public function with(ContextKey $key, $value): self
-    {
-        return new self($key, $value, $this);
-    }
+    // Optimization for spans to avoid copying the context array.
+    private static ContextKey $spanContextKey;
+    private ?SpanInterface $span = null;
+    /** @var array<int, mixed> */
+    private array $context = [];
+    /** @var array<int, ContextKey> */
+    private array $contextKeys = [];
 
-    /**
-     * @todo: Implement this on the API side
-     */
-    public function withContextValue(ImplicitContextKeyedInterface $value): self
-    {
-        return $value->storeInContext($this);
-    }
-
-    /**
-     * Makes `$this` the currently active {@see Context}.
-     *
-     * @todo: Implement this on the API side
-     */
-    public function activate(): ScopeInterface
+    private function __construct()
     {
-        return self::storage()->attach($this);
+        self::$spanContextKey = SpanContextKey::instance();
```

Comment:
Should this use `class_exists()` or should the context key be moved to `open-telemetry/context`?

Reply:
IMO, move to open-telemetry/context if it's not too invasive.